File: 2021-remove-all-occurrences-of-a-substring/2021-remove-all-occurrences-of-a-substring.cs
public class Solution {
    public string RemoveOccurrences(string s, string part) {
        var n = s.Length;
        var p = part.Length;
        var stringBuilder = new StringBuilder();

        for (var i = 0; i < n; i++) {
            stringBuilder.Append(s[i]);
            var remove = true;
            if (stringBuilder.Length < p)
                continue;

            for (var j = 1; j <= p; j++) {
                if (stringBuilder[^j] == part[^j])
                    continue;
                remove = false;
                break;
            }

            if (remove)
                stringBuilder.Length -= p;
        }

        return stringBuilder.ToString();
    }
}

File: 2021-remove-all-occurrences-of-a-substring/2021-remove-all-occurrences-of-a-substring.go
func removeOccurrences(s string, part string) string {
	for i := 0; i < len(s); i++ {
		if s[i] == part[0] {
			if i + len(part) < len(s) {
				if s[i : i + len(part)] == part {
					s = s[: i] + s[i + len(part) : ]
					i = -1
				}
			} else {
				if s[i: ] == part {
					s = s[ :i]
				}
				break
			}
		}
	}

	return s
}